common/persistence/cassandraMetadataPersistence.go

```diff
@@ -139,23 +147,30 @@ func (m *cassandraMetadataPersistence) CreateDomain(request *CreateDomainRequest
 		request.Status,
 		request.Description,
 		request.OwnerEmail,
-		request.Retention,
-		request.EmitMetric)
+		request.Config.Retention,
+		request.Config.EmitMetric,
+		request.ReplicationConfig.ActiveClusterName,
+		request.ReplicationConfig.FailoverVersion,
+		clusterReplicationConfigs,
+	)
 
 	previous := make(map[string]interface{})
 	applied, err := query.MapScanCAS(previous)
+
+	if !applied || err != nil {
+		// Domain already exist.  Delete orphan domain record before returning back to user
+		if errDelete := m.session.Query(templateDeleteDomainQuery, domainUUID).Exec(); errDelete != nil {
```

Comment:
It is dangerous to delete the domain on any errors.

Reply:
this is the create API, and existing logic does the same thing, except existing logic have bug dealing with second create call to Cassandra having errors.

Reply:
fixed